src/adaptation/ecap/XactionRep.cc

```diff
@@ -241,12 +242,8 @@ Adaptation::Ecap::XactionRep::start()
         adaptHistoryId = ah->recordXactStart(service().cfg().key, current_time, false);
         SBuf matched;
         for (auto h: Adaptation::Config::metaHeaders) {
```

Comment:
I agree that there is a problem and think we need to fix by creating Adaptation::History not only if Log::TheConfig.hasAdaptToken is true but also on non-empty Adaptation::Config::metaHeaders (for otherwise, we cannot 'execute' on nil history). So the solution would be:

```
Adaptation::History::Pointer
HttpRequest::adaptLogHistory() const
{
    return HttpRequest::adaptHistory(Log::TheConfig.hasAdaptToken || !Adaptation::Config::metaHeaders.empty());
}
```

Reply:
As discussed [earlier](https://github.com/measurement-factory/squid/pull/36#issuecomment-504640142), this should probably be simplified to remove all conditions. Otherwise, the history may be missing for dynamically loaded error pages that use logformat %code. The distinction between history for logging purposes and history for meta header exchanges will be gone.

Long-term, we should load _all_ error pages at (re)configuration time (and re-add a complex condition that takes error pages into account) and/or optimize header copying so that it becomes a very fast operation.

The PR description should disclose and justify this performance regression as well as long-term optimization plans.

Reply:
I applied the suggested [fix](https://github.com/measurement-factory/squid/pull/36/files#r300169419) and added a TODO for the 'dynamically loaded error pages' problem.